core/common/src/test/java/tachyon/wire/BlockInfoTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.wire;
+
+import java.util.List;
+import java.util.Random;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.google.common.collect.Lists;
+
+public class BlockInfoTest {
```

Comment:
I see. Removed the `@JsonProperty` annotation.

If I understand you correctly, the alternative base on reflection would consist of a single test class with a list of the wire type classes. The test class would iterate over all the test classes running the JSON and Thrift tests. Each wire type class would need to provide the createExample and also checkEquality methods.

I think I prefer having one test class per wire type. It requires a bit more boilerplate, but it avoids having to put createExample and checkEquality into the production code.